### PR TITLE
[SPARK-8669] [SQL] Fix crash with BINARY (ENUM) fields with Parquet 1.7

### PR DESCRIPTION
**Pull request by stshe**
Patch to fix crash with BINARY fields with ENUM original types.

### CODE REVIEW

**Comment by yhuai** [comment]
test this please.

**Comment by yhuai** [comment]
Can you update the title to `[SPARK-8669] [SQL] Fix crash with BINARY (ENUM) fields with Parquet 1.7`

**Comment by liancheng** [comment]
Thanks for fixing this! `ENUM` is not specified in the Parquet format spec, but according to parquet-mr 1.7.0, it is only used for converting Avro, ProtoBuf, and Thrift files. Double checked that in all cases `ENUM` is mapped to UTF8 string. So I think it's OK to always map `ENUM` to `StringType` regardless the value of `assumeBinaryIsString`.

File: sql/core/src/main/scala/org/apache/spark/sql/parquet/CatalystSchemaConverter.scala [hunk 1]
```diff
@@ -178,8 +178,8 @@ private[parquet] class CatalystSchemaConverter(
       case BINARY =>
         field.getOriginalType match {
           case UTF8 => StringType
-          case null if assumeBinaryIsString => StringType
-          case null => BinaryType
+          case null | ENUM if assumeBinaryIsString => StringType
+          case null | ENUM => BinaryType
```
Comment (liancheng) [inline sql/core/src/main/scala/org/apache/spark/sql/parquet/CatalystSchemaConverter.scala]: Let's revert changes to these two lines and change the first `case` branch to:

```scala
case UTF8 | ENUM => StringType
```

**Comment by stevencanopy** [comment]
Thanks @liancheng for looking into the Parquet spec! Yes, I first encountered the bug when I was converting Avro to Parquet with parquet-mr 1.7.0.

I've reverted the change to the two null cases, added ENUM to the UTF8 case, and updated the test case.

**Comment by liancheng** [comment]
ok to test

**Comment by liancheng** [comment]
Hm, not knowing why @SparkQA said the build failed, it was actually a success. Anyway, I'm merging this to master. Thanks again for working on this!

**Comment by liancheng** [comment]
Correction: the previous Jenkins build terminated prematurely, but @AmplabJenkins didn't mark it as a failure. Re-triggering the build.

**Comment by liancheng** [comment]
retest this please

**Comment by liancheng** [comment]
This LGTM, pending Jenkins build result.

**Comment by marmbrus** [comment]
Thanks! Merging to master.

**Pull request closed without merging**